### DISCUSSION TITLE
Problem z uruchomieniem testów.

### DISCUSSION THREAD

**Discussion by mystertaran** [przeprogramowani/advent-of-frontend#16] [Pierwsze kroki] [answered]
Cześć wszystkim.

Zacząłem niestety kalendarz adwentowy z małą obsuwą, wiec dopiero od dzisiaj. Utknąłem przy uruchamianiu testów, ponieważ terminal krzyczy, że nie ma zdefiniowanych testów:

> $ npm test
> 
>  advent-of-frontend@0.0.1 test
>  jest ./tasks/$(date +'%Y-%m-%d')/index.test.ts
> 
>   Invalid testPattern .\\tasks\\$(date|+'%Y-%m-%d')\\index.test.ts supplied. Running all tests instead.
>   Invalid testPattern .\\tasks\\$(date|+'%Y-%m-%d')\\index.test.ts supplied. Running all tests instead.
>  PASS  tasks/2023-11-27/index.test.ts
>  PASS  tasks/2023-11-26/index.test.ts
>  FAIL  tasks/2023-12-05/index.test.ts
>   ● Test suite failed to run
> 
>     Your test suite must contain at least one test.
> 
>       at onResult (node_modules/@jest/core/build/TestScheduler.js:133:18)
>       at node_modules/@jest/core/build/TestScheduler.js:254:19
>       at node_modules/emittery/index.js:363:13
>           at Array.map (<anonymous>)
>       at Emittery.emit (node_modules/emittery/index.js:361:23)
> 
> Test Suites: 1 failed, 2 passed, 3 total
> Tests:       2 passed, 2 total
> Snapshots:   0 total
> Time:        2.999 s, estimated 3 s
> Ran all test suites.

Ktoś miał podobny problem? npm install oczywiście wykonany, wszystko up-to-date (wersja noda switchnięta na 18.0.0). 

@EDIT
Wiem gdzie był problem - z pierwszej linijki usunąłem komentarz "// Tutaj skopiuj testy dla zadania. Uruchom je poleceniem `npm test`" i przez to nie przechodziło testów z powyższym błędem.

Dodatkowo zapytam, czy da się cofnąć do poprzednich dni z zadaniami?

Z góry dziękuje za odpowiedzi!

**Comment by lotyszm** [comment, marked as answer]
Lista zadań:

- https://www.opanujfrontend.pl/advent/2023-12-01
- https://www.opanujfrontend.pl/advent/2023-12-02
- .... 
- https://www.opanujfrontend.pl/advent/2023-12-05

`npm run create` tworzy katalog w lokalizacji  `./tasks/YYYY-mm-dd`   z aktualną datą.

![image](https://github.com/przeprogramowani/advent-of-frontend/assets/2004150/623a11aa-b607-4525-bde0-a4a069833dc3)


`npm test` uruchamia test z dzisiejszej datą, ale możesz odpalić test dla wybranej daty `npx jest ./tasks/2023-12-01/index.test.ts`

Możesz też uruchomić wszystkie testy `npm run test:all` z katalogu głównego aplikacji.

**Comment by mystertaran** [comment]
Siemka,

Dzięki za wytłumaczenie co do starych zadań i wywoływania testów na każdy
dzień.

Czy jak zrobiłem fork repozytorium to commity i pushe będą liczyły się do
statystyk na githubie?

Pozdro,
Szymon

W dniu wt., 5.12.2023 o 14:16 mlotysz ***@***.***> napisał(a):

**Comment by lotyszm** [comment]
Chodzi Ci o Contributions calendar? 

Commity do statystyk GitHuba zliczą się z datą utworzenia commita. 

Czyli jeśli dziś zrobisz zadania z poprzednich dni i dziś dasz commit, to statystyka będzie też z dzisiejszą data.

**Comment by mystertaran** [comment]
Chodzi o to, że dałem z 3 commity przy robieniu tego taska, może 4, a na
profilu u siebie widziałem jedną aktywność i to pewnie było stworzenie repo
poprzez fork.

W dniu wt., 5.12.2023 o 14:25 mlotysz ***@***.***> napisał(a):

**Reply by lotyszm** [reply]
Trudno mi powiedzieć. Może ktoś inny się wypowie jak dokładnie działa ta statystyka w contributions calendar.

**Reply by psmyrdek** [reply]
Tutaj więcej na ten temat - https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile
